go-controller/pkg/ovn/master.go

```diff
@@ -549,29 +549,81 @@ func (oc *Controller) SetupMaster(masterNodeName string) error {
 	return nil
 }
 
-func addNodeLogicalSwitchPort(logicalSwitch, portName, portType, addresses, options string) (string, error) {
-	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-type", portName, portType,
-		"--", "lsp-set-options", portName, options,
-		"--", "lsp-set-addresses", portName, addresses)
+func addNodeLogicalSwitchPort(ovnNBClient goovn.Client, logicalSwitch, portName, portType, addresses, options string) (string, error) {
+	var cmds []*goovn.OvnCommand
+	var goOvnOptions map[string]string
+
+	lsp, err := ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
```

Comment:
how is it possible that the cache doesn't have the LSP table? If the table isn't in the cache how are we going to add a port to the table that doesn't exist?

Reply:
@trozet Good point. Just followed the pattern here https://github.com/ovn-org/ovn-kubernetes/blob/b77e0e3027d905089daa50bcd2731e9b4498079e/go-controller/pkg/ovn/pods.go#L268

I see why this will be a problem. Will fix the error checking.